runtime/plugin/news.lua

```diff
@@ -0,0 +1,53 @@
+-- skip when nvim is a release build
+if vim.version().prerelease ~= "dev" then
+  return
+end
+
+-- skip if user code explicitly disables
+if vim.g.news_check == false then
+  return
+end
+
+-- skip in embedded situations
+if vim.g.vscode or vim.g.started_by_firenvim then
```

Comment:
can check for embedding by seeing if the TUI is (not) running. see code example at :help nvim_list_uis: https://github.com/neovim/neovim/blob/47866cd8d20c62afa8a3c3929d3aada2db9162f5/runtime/doc/api.txt#L1072-L1074